src/main/java/com/aws/greengrass/config/PlatformResolver.java

```diff
@@ -94,22 +94,22 @@ public PlatformResolver(DeviceConfiguration deviceConfiguration) {
         return platform;
     }
 
+    /**
+     * Get the OS type for this system.
+     * 
+     * @return one of our supported identified OSes
+     */
     @SuppressFBWarnings("DMI_HARDCODED_ABSOLUTE_FILENAME")
-    private static String getOSInfo() {
+    public static String getOSInfo() {
         if (isWindows) {
             return OS_WINDOWS;
         }
-        try {
-            String sysver = Exec.sh("uname -a").toLowerCase();
-
-            if (sysver.contains("darwin")) {
-                return OS_DARWIN;
-            }
-            if (Files.exists(Paths.get("/proc"))) {
-                return OS_LINUX;
-            }
-        } catch (IOException | InterruptedException e) {
-            logger.error("Error trying to determine OS", e);
+        String osName = System.getProperty("os.name").toLowerCase();
```

Comment:
Do you know how the `os.name` property is set?

Reply:
It is builtin to the JVM.

Reply:
https://github.com/openjdk/jdk/blob/9604ee82690f89320614b37bfef4178abc869777/src/java.base/macosx/native/libjava/java_props_macosx.c#L234-L235

https://github.com/openjdk/jdk/blob/9604ee82690f89320614b37bfef4178abc869777/src/java.base/unix/native/libjava/java_props_md.c#L403